packages/apps/reputation-oracle/server/src/modules/cron-job/cron-job.service.ts

```diff
@@ -85,91 +100,220 @@ export class CronJobService {
 
   @Cron('*/2 * * * *')
   /**
-   * Process a pending webhook job.
-   * @returns {Promise<void>} - Returns a promise that resolves when the operation is complete.
+   * Processes all pending incoming webhooks, marking them as completed upon success.
+   * Handles any errors by logging them and updating the webhook status.
+   * @returns {Promise<void>} A promise that resolves when all pending incoming webhooks have been processed.
    */
-  public async processPendingWebhooks(): Promise<void> {
+  public async processPendingIncomingWebhooks(): Promise<void> {
     const isCronJobRunning = await this.isCronJobRunning(
-      CronJobType.ProcessPendingWebhook,
+      CronJobType.ProcessPendingIncomingWebhook,
     );
 
     if (isCronJobRunning) {
       return;
     }
 
-    this.logger.log('Pending webhooks START');
-    const cronJob = await this.startCronJob(CronJobType.ProcessPendingWebhook);
+    this.logger.log('Pending incoming webhooks START');
+    const cronJob = await this.startCronJob(
+      CronJobType.ProcessPendingIncomingWebhook,
+    );
 
     try {
-      const webhookEntities = await this.webhookRepository.findByStatus(
-        WebhookStatus.PENDING,
+      const webhookEntities = await this.webhookIncomingRepository.findByStatus(
+        WebhookIncomingStatus.PENDING,
       );
 
       for (const webhookEntity of webhookEntities) {
-        let resultsUrl;
         try {
           const { chainId, escrowAddress } = webhookEntity;
-          resultsUrl = await this.payoutService.executePayouts(
+
+          await this.escrowCompletionTrackingService.createEscrowCompletionTracking(
             chainId,
             escrowAddress,
           );
+
+          webhookEntity.status = WebhookIncomingStatus.COMPLETED;
+          await this.webhookIncomingRepository.updateOne(webhookEntity);
         } catch (err) {
-          this.logger.error(`Error sending webhook: ${err.message}`);
-          await this.webhookService.handleWebhookError(webhookEntity);
-          continue;
+          if (err instanceof DatabaseError) {
+            if (
+              (err as DatabaseError).message.includes(
```

Comment:
You know, `BaseRepository` service has `handleQueryFailedError` in each method: 
```
async createUnique(item: T): Promise<T> {
    try {
      await this.insert(item);
    } catch (error) {
      if (error instanceof QueryFailedError) {
        throw handleQueryFailedError(error);
      } else {
        throw error;
      }
    }
    return item;
  }
```

It return `new DatabaseError(...)`.
On the top level, for example in `CronService` we can handle it as common database error and check only `err instanceof DatabaseError`.

Reply:
@eugenvoronov `DatabaseError` can also come from `updateOne` and other methods of `BaseRepository`, also it can happen due to different reasons (not only duplication), so checking only the instance is wrong

Also if we have some sort of utility - it will be easier to update the code in case if we change `DatabaseError`: you will need to handle only one place

Reply:
I Implemented `isDuplicatedError` in database utils.